### ISSUE TITLE
Implement Token Shortcuts

### ISSUE THREAD

**Issue by JamesSmartCell** [AlphaWallet/alpha-wallet-android#3020]
Add a shortcut icon to the homepages which takes you straight to a token and specific account when clicked on eg:

<Homepage Icon> -> wallet 0xabcdef0123456789... token: Satoshi's Villa NFT

This should be an option on each token's page. The UI hasn't been designed.

This is very useful for utility tokens, eg TokenScript door control.

**Comment by hboon** [comment]
iOS lets apps "donate shortcuts", basically the app provides a JSON payload, the OS shows a UI for it and when the user taps it, the app can read the payload and do something. The shortcut can show up in search results like in the screenshot below, or the OS can display it intelligently by inferring the best time and place (eg. it might show the door token if the shortcut is often donated by the app at a certain location or used at 8am in the morning).

See https://github.com/AlphaWallet/alpha-wallet-ios/pull/3437

Does Android have something like that?

**Comment by JamesSmartCell** [comment]
> iOS lets apps "donate shortcuts", basically the app provides a JSON payload, the OS shows a UI for it and when the user taps it, the app can read the payload and do something. The shortcut can show up in search results like in the screenshot below, or the OS can display it intelligently by inferring the best time and place (eg. it might show the door token if the shortcut is often donated by the app at a certain location or used at 8am in the morning).
> 
> See [AlphaWallet/alpha-wallet-ios#3437](https://github.com/AlphaWallet/alpha-wallet-ios/pull/3437)
> 
> Does Android have something like that?

Yes! it does: https://developer.android.com/develop/ui/views/launch/shortcuts

It works in a neat way too. This feature would be extremely useful.

**Comment by JamesSmartCell** [comment]
@seabornlee would you be able to pick this one up? I think it would be invaluable for impromptu TokenScript demos at the very least. I was demoing my home TokenScript and the office one recently and it wasn't a good demo as I had AW set up using Testnet on a different wallet. 

So the sequence was:

- Open AW.
- See I wasn't on the right wallet, click through settings, find the right wallet and change.
- Change from testnet to mainnet.
- Locate token and click on it.
- Use Open verb.

Meanwhile a dedicated smarthome app would have been there in about 3 easy clicks.

With this system I should be able to go:

- Use token shortcut to take me straight to the right wallet and token.
- Click 'open' and I'm in.

Ideally, you wouldn't change the existing set wallet, so after the demo/arriving home the user can go back to work and be in the original wallet/testnet setup. This should be easier actually as the ```getIntentData()``` in ```NFTActivity``` uses the supplied wallet and token.

**Comment by seabornlee** [comment]
**Tasks**
Spike shortcuts
Click shortcuts and open token in inactive wallet
Create shortcuts with same token in multiple wallets
